src/full/Agda/TypeChecking/Rules/Term.hs

```diff
@@ -1572,20 +1571,21 @@ inferExprForWith (Arg info e) = verboseBracket "tc.with.infer" 20 "inferExprForW
             , "context size = " <+> do text . show =<< getContextSize
             , "current context = " <+> do prettyTCM =<< getContextTelescope
             ]
-          typeError $ WithOnFreeVariable e v0
+          typeError $ WithOnFreeVariable e v
         _        -> return ()
       -- Possibly insert hidden arguments.
       TelV tel t0 <- telViewUpTo' (-1) (not . visible) t
-      case unEl t0 of
+      (v, t) <- case unEl t0 of
         Def d vs -> do
           isDataOrRecordType d >>= \case
             Nothing -> return (v, t)
             Just{}  -> do
               (args, t1) <- implicitArgs (-1) notVisible t
-              -- #6868: trigger instance search if we inserted any instance arguments
-              when (any isInstance args) $ solveAwakeInstanceConstraints
               return (v `apply` args, t1)
         _ -> return (v, t)
```

Comment:
This `v` is now more reduced than before, is it?  (A priori that could affect some error messages, maybe that was why there was a `v0`...)

Reply:
Yes, but I checked all the places where this function is used (which is a grand total of 2) and in both places we continue by normalizing it. So returning the reduced version here is fine, and it could save a lot of work if computing the whnf is expensive.

Reply:
I can confirm.